### PR TITLE
[msbuild] Improve the error message when the SupportedOSPlatformVersion is lower than the minimum. Fixes #21368.

### LINKED ISSUE #21368: Show a better error message when SupportedOSPlatformVersion is too low
Something better than this:

image

### PR DESCRIPTION
**Pull request by rolfbjarne**
This isn't very user friendly:

    ILLink : unknown error IL7000: An error occurred while executing the custom linker steps. Please review the build log for more information.
    ILLINK : error MT0073: Microsoft.iOS 18.0.8337 does not support a deployment target of 10.0 for iOS (the minimum is 11.0). Please select a newer deployment target in your project's Info.plist or change the SupportedOSPlatformVersion property in your project file.
    ILLINK : error MT2301: The linker step 'Setup' failed during processing: Microsoft.iOS 18.0.8337 does not support a deployment target of 10.0 for iOS (the minimum is 11.0). Please select a newer deployment target in your project's Info.plist or change the SupportedOSPlatformVersion property in your project file.
    [...]/packages/microsoft.net.illink.tasks/8.0.8/build/Microsoft.NET.ILLink.targets(87,5): error NETSDK1144: Optimizing assemblies for size failed. Optimization can be disabled by setting the PublishTrimmed property to false.

So improve this to only show a single error message:

    The SupportedOSPlatformVersion value '10.0' in the project file is lower than the minimum value '11.0'.

Fixes https://github.com/xamarin/xamarin-macios/issues/21368.

### CODE REVIEW

**Comment by rolfbjarne** [comment]
CC @PureWeen

**Review by emaf — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

File: msbuild/Xamarin.Localization.MSBuild/MSBStrings.resx [hunk 1]
```diff
@@ -1576,4 +1576,17 @@
         <comment>RuntimeIdentifier: don't translate (it's the name of a MSBuild property)</comment>
     </data>
 
+    <data name="E7125" xml:space="preserve">
+        <value>Unable to parse the value '{0}' for the property '{1}.</value>
+        <comment>
+            {0}: user-provided value
+            {1]: name of an MSBuild property
```
Comment (stephen-hawley) [inline msbuild/Xamarin.Localization.MSBuild/MSBStrings.resx:1583]: You probably want this to be `{1}` and not `{1]`

**Review by stephen-hawley — CHANGES_REQUESTED** [review]

File: msbuild/Xamarin.Localization.MSBuild/MSBStrings.resx (hunk 1 above)
Comment (rolfbjarne) [reply msbuild/Xamarin.Localization.MSBuild/MSBStrings.resx]: @stephen-hawley Fixed!

**Review by stephen-hawley — APPROVED** [review]

**Review by mandel-macaque — APPROVED** [review]

**Pull request merged**